services/resource-usage-tracker/tests/unit/with_dbs/test_background_task.py

```diff
@@ -0,0 +1,179 @@
+            resource_tracker_service_runs.insert().values(
+                **random_resource_tracker_service_run(
+                    service_run_id=_SERVICE_RUN_ID_S4L_10_MIN_OLD,
+                    product_name="s4l",
+                    last_heartbeat_at=_LAST_HEARTBEAT_10_MIN_OLD,
+                    started_at=_LAST_HEARTBEAT_10_MIN_OLD - timedelta(minutes=1),
+                )
+            )
+        )
+        con.execute(
+            resource_tracker_service_runs.insert().values(
+                **random_resource_tracker_service_run(
+                    service_run_id=_SERVICE_RUN_ID_OSPARC_NOW,
+                    product_name="osparc",
+                    last_heartbeat_at=_LAST_HEARTBEAT_NOW,
+                )
+            )
+        )
+        # Populate credit transactions table
+        con.execute(
+            resource_tracker_credit_transactions.insert().values(
+                **random_resource_tracker_credit_transactions(
+                    service_run_id=_SERVICE_RUN_ID_OSPARC_10_MIN_OLD,
+                    product_name="osparc",
+                    last_heartbeat_at=_LAST_HEARTBEAT_10_MIN_OLD,
+                    transaction_status="PENDING",
+                )
+            )
+        )
+        con.execute(
+            resource_tracker_credit_transactions.insert().values(
+                **random_resource_tracker_credit_transactions(
+                    service_run_id=_SERVICE_RUN_ID_S4L_10_MIN_OLD,
+                    product_name="s4l",
+                    last_heartbeat_at=_LAST_HEARTBEAT_10_MIN_OLD,
+                    transaction_status="PENDING",
+                )
+            )
+        )
+        con.execute(
+            resource_tracker_credit_transactions.insert().values(
+                **random_resource_tracker_credit_transactions(
+                    service_run_id=_SERVICE_RUN_ID_OSPARC_NOW,
+                    product_name="osparc",
+                    last_heartbeat_at=_LAST_HEARTBEAT_NOW,
+                    transaction_status="PENDING",
+                )
+            )
+        )
+
+        yield
+
+        con.execute(resource_tracker_credit_transactions.delete())
+        con.execute(resource_tracker_service_runs.delete())
+
+
+async def test_process_event_functions(
+    rabbitmq_client: Callable[[str], RabbitMQClient],
+    mocked_redis_server: None,
+    postgres_db: sa.engine.Engine,
+    resource_tracker_setup_db,
+    initialized_app,
+):
+    engine = initialized_app.state.engine
+    app_settings: ApplicationSettings = initialized_app.state.settings
+    resource_tracker_repo: ResourceTrackerRepository = ResourceTrackerRepository(
+        db_engine=engine
+    )
+
+    for _ in range(app_settings.RESOURCE_USAGE_TRACKER_MISSED_HEARTBEAT_COUNTER_FAIL):
+        await periodic_check_of_running_services_task(initialized_app)
+
+    # Check max acceptable missed heartbeats reached before considering them as unhealthy
+    with postgres_db.connect() as con:
```

Comment:
Can you perhaps move these away as fixtures and make sure you clean up after you fill db tables
Not only it makes it more difficult to read but also I think is the reason why you have to delete tables on the fixture above

Reply:
As mentioned above, all tables were already empty and the deletion was not needed there. Also, I am not sure I get the fixture, because this is part of the test to check the state of database in the end whether all values are correct.